kedro/contrib/io/matplotlib/matplotlib_s3_writer.py

```diff
@@ -0,0 +1,162 @@
+        from kedro.contrib.io.matplotlib_s3_writer import MatplotlibWriterS3
+
+        # Saving single plot
+        plt.plot([1, 2, 3], [4, 5, 6])
+        single_plot_writer = MatplotlibWriterS3(
+            bucket="my-super-great-bucket", filepath="matplot_lib_single_plot.png"
+        )
+        single_plot_writer.save(plt)
+
+        # Saving dictionary of plots (with SSE)
+        plots_dict = {}
+        for colour in ["blue", "green", "red"]:
+            plots_dict[colour] = plt.figure()
+            plt.plot([1, 2, 3], [4, 5, 6], color=colour)
+            plt.close()
+        dict_plot_writer = MatplotlibWriterS3(
+            bucket="my-super-great-bucket", filepath="matplotlib_dict",
+        )
+        dict_plot_writer.save(plots_dict)
+
+        # Saving list of plots
+        plots_list = []
+        for index in range(5):
+            plots_list.append(plt.figure())
+            plt.plot([1,2,3],[4,5,6], color=colour)
+        list_plot_writer = MatplotlibWriterS3(
+            bucket="my-super-great-bucket", filepath="matplotlib_list"
+        )
+        list_plot_writer.save(plots_list)
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        bucket_name: str,
+        filepath: str,
+        s3fs_args: Optional[Dict] = None,
+        credentials: Optional[Dict[str, Any]] = None,
+        savefig_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``MatplotlibWriter``.
+
+        Args:
+            bucket_name: Name of the bucket without "s3://" prefix
+            filepath: Path to a matplot object file.
+            s3fs_args: Dictionary of arguments
+            credentials: A dictionary of s3 access and secret keys.
+                Must contain ``aws_access_key_id`` and ``aws_secret_access_key``.
+                Updates ``s3_client_args`` if provided.
+            savefig_args: Save args passed to `plt.savefig`. See
+                https://matplotlib.org/api/_as_gen/matplotlib.pyplot.savefig.html
+        """
+
+        _credentials = copy.deepcopy(credentials) or {}
+
+        self._s3fs_args = s3fs_args or {}
+
+        if _credentials:
+            if "client_kwargs" not in self._s3fs_args.keys():
+                self._s3fs_args["client_kwargs"] = {}
+
+            self._s3fs_args["client_kwargs"]["aws_access_key_id"] = _credentials[
+                "aws_access_key_id"
+            ]
+            self._s3fs_args["client_kwargs"]["aws_secret_access_key"] = _credentials[
+                "aws_secret_access_key"
+            ]
+
+        self._filepath = filepath
+        self._savefig_args = savefig_args if savefig_args else dict()
+        self._bucket_name = bucket_name
+
+        _s3 = S3FileSystem(**self._s3fs_args)
+        self._s3 = _s3
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            bucket_name=self._bucket_name,
+            filepath=self._filepath,
+            s3fs_args=self._s3fs_args,
+        )
```

Comment:
This is exposing sensitive information, like `aws_secret_access_key`, via public API (`describe`), which is not a good idea! I would remove this from the description and add the `save_args` instead.

Reply:
Depending on your response below

Reply:
No longer relevant since keys are passed to `credentials` not `s3fs_args`